api/v1beta1/trex_performance_types.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TrexPerformanceSpec defines the desired state of TrexPerformance
+type TrexPerformanceSpec struct {
+	// +kubebuilder:validation:Required
+	// Name of a workflow step. The step name will be used for example to create
+	// a logs directory.
+	StepName string `json:"stepName"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="local-storage"
```

Comment:
Can you please add the following for each parameter and then run `make bundle`? The command will update `config/manifests/bases/test-operator.clusterserviceversion.yaml`. The updated version of this file should be part of the PR.

```
// +operator-sdk:csv:customresourcedefinitions:type=spec
```